### ISSUE TITLE
Not reporting layer print time

### ISSUE THREAD

**Issue by Astroman** [foosel/OctoPrint#1481]
In Version 1.2.15, layer print time on the gcode viewer page is always reported as "a few seconds." Remaining print time, etc seems ok though. If you can't see this problem yourself, let me know and I'll be glad to provide whatever other info you need. Thanks.

**Comment by GitIssueBot** [comment]
Hi @Astroman,

It looks like there is some **information missing** from your bug report that will be needed in order to solve the problem. Read the [Contribution Guidelines](https://github.com/foosel/OctoPrint/blob/master/CONTRIBUTING.md) which will provide you with a template to fill out here so that your bug report is ready to be investigated (I promise I'll go away then too!).

If you did not intend to report a bug but wanted to **request a feature or brain storm** about some kind of development, please take special note of the title format to use as described in the [Contribution Guidelines](https://github.com/foosel/OctoPrint/blob/master/CONTRIBUTING.md).

**Please do not abuse the bug tracker as a support forum** - if you have a question or otherwise need some kind of help or support refer to the [Mailinglist](https://groups.google.com/group/octoprint) or the [G+ Community](https://plus.google.com/communities/102771308349328485741) instead of here.

Also **make sure you are at the right place** - this is the bug tracker of the official version of OctoPrint, not the Raspberry Pi image OctoPi nor any unbundled third party OctoPrint plugins or unofficial versions. Make sure too that you have **read through the [Frequently Asked Questions](https://github.com/foosel/OctoPrint/wiki/FAQ)** and searched the [**existing tickets**](https://github.com/foosel/OctoPrint/search?q=&ref=cmdform&type=Issues) for your problem - try multiple search terms please.

I'm marking this one now as needing some more information. Please understand that if you do not provide that information within the next two weeks (until 2016-09-20 17:20 UTC) I'll close this ticket so it doesn't clutter the bug tracker. This is nothing personal, so please just be considerate and help the maintainers solve this problem quickly by following the guidelines linked above. Remember, the less time the devs have to spend running after information on tickets, the more time they have to actually solve problems and add awesome new features. Thank you!

Best regards,
~ Your friendly GitIssueBot

PS: I'm just an automated script, not a human being, so don't expect any replies from me :) Your ticket is read by humans too, I'm just not one of them.

**Comment by Astroman** [comment]
#### What were you doing?

Printing an object and looking at the Gcode viewer

#### What did you expect to happen?
See actual layer print time
#### What happened instead?
Layer print time always showed "a few seconds" even though it was much longer
#### Branch & Commit or Version of OctoPrint
1.2.15 (master branch)
[Can be found in the lower left corner of the web interface.]

#### Printer model & used firmware incl. version
Printrbot simple, not sure about version
[If applicable, always include if unsure.]

#### Browser and Version of Browser, Operating System running Browser
Safari 9.1.3, Mac OS 10.11.6
[If applicable, always include if unsure.]

#### Link to octoprint.log

[On gist.github.com or pastebin.com. Always include and never truncate.]

#### Link to contents of terminal tab or serial.log

[On gist.github.com or pastebin.com. If applicable, always include if unsure or
reporting communication issues. Never truncate.]

#### Link to contents of Javascript console in the browser
Not printing right now, will provide later if necessary
[On gist.github.com or pastebin.com or alternatively a screenshot. If applicable -
always include if unsure or reporting UI issues.]

#### Screenshot(s) showing the problem:
Not printing right now, will provide later if necessary
[If applicable. Always include if unsure or reporting UI issues.]

**Comment by foosel** [comment]
> Layer print time always showed "a few seconds" even though it was much longer

Layer print times - like everything else - are estimates which depending on multiple factors can be more or less accurate. For this reason I switched all print time estimates in 1.2.14/15 to be displayed in a very fuzzy way since people do depend too much on them to be accurate otherwise.

On the `maintenance` branch/future 1.2.16 release I've switched at least the individual layer times in the GCODE viewer back to show as full durations instead of fuzzed up estimates:

![image](https://cloud.githubusercontent.com/assets/83657/18381144/189b5482-767b-11e6-8fde-3a236c8b7ed0.png)

That's all I can do here for now. 

If the layer times in general are off (it says "a couple of seconds" but takes several minutes instead), that's most certainly caused by something in your GCODE files making the estimate in the GCODE viewer go haywire. I'd need an example file to be able to look further into this.

PS: Always copy-paste the *full* bug template. Do not remove anything from it that is not enclosed in [ and ]. Otherwise the bot will get angry/stay angry.

**Comment by foosel** [comment]
@Astroman 

> If the layer times in general are off (it says "a couple of seconds" but takes several minutes instead), that's most certainly caused by something in your GCODE files making the estimate in the GCODE viewer go haywire. I'd need an example file to be able to look further into this.

Is this the case or were you only referring to the layer times being stated in a fuzzy manner? Important to know, because if the former there might be more action needed, if the latter this can be considered closed.

**Comment by Astroman** [comment]
I looked more carefully at the issue and it appears to say “a few seconds” whenever the layer print time is under a minute. In other words, it never estimates short layer times in seconds. I thought it always said “a few seconds" since I happened to be looking at the gcode viewer only when layer times were under a minute.

That said, I think it used to provide estimates for layer times that were under a minute. Of course, I can’t confirm this, and if this is not correct, I apologize. I seem to remember using these relatively short layer print times of 20-50 seconds or so to estimate the amount of time left to get to a higher layer.

I wonder if you are rounding off or truncating to the nearest minute which would be fine for long layer times but misleading for short ones, espeially those under a minute.

Thanks.

—Dave
